apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/Rates/index.jsx

```diff
@@ -66,24 +67,65 @@ export default function ProviderRate({
 
   const providerRef = useRef(null);
 
+  const getDefaultPartner = useCallback(
+    rate => {
+      const defaultPartner = rates?.find((rate, index) => {
+        return filter.every(item => [FILTER.centralised, rate.tradeMethod].includes(item));
+      });
+      const defaultDexPartner = DEX_PROVIDERS.find((rate, index) => {
+        return filter.every(item => [FILTER.decentralised, FILTER.float].includes(item));
+      });
+      if (defaultPartner) {
+        return defaultPartner;
+      } else if (decentralizedSwapAvailable && defaultDexPartner) {
+        return defaultDexPartner;
+      } else {
+        return {};
+      }
+    },
```

Comment:
This seems unnecessarily complex. I'm sure you can simplify it, but perhaps I'm lacking context so I'll leave the approval to someone else. Same applies to [the logic inside the ProviderRate ](https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/Rates/Filter.jsx#L40-L69)components for that matter.

Reply:
You are right. Now the code is simpler :smiley: